src/community/smart-data-loader/src/test/java/org/geoserver/smartdataloader/data/store/EnvParametrizationStoreTest.java

```diff
@@ -0,0 +1,142 @@
+package org.geoserver.smartdataloader.data.store;
+
+import static org.geotools.data.postgis.PostgisNGDataStoreFactory.SSL_MODE;
+import static org.geotools.jdbc.JDBCDataStoreFactory.DATABASE;
+import static org.geotools.jdbc.JDBCDataStoreFactory.DBTYPE;
+import static org.geotools.jdbc.JDBCDataStoreFactory.HOST;
+import static org.geotools.jdbc.JDBCDataStoreFactory.PASSWD;
+import static org.geotools.jdbc.JDBCDataStoreFactory.PORT;
+import static org.geotools.jdbc.JDBCDataStoreFactory.USER;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.DataStoreInfo;
+import org.geoserver.catalog.WorkspaceInfo;
+import org.geoserver.platform.GeoServerEnvironment;
+import org.geoserver.smartdataloader.AbstractJDBCSmartDataLoaderTestSupport;
+import org.geoserver.smartdataloader.JDBCFixtureHelper;
+import org.geoserver.smartdataloader.data.SmartDataLoaderDataAccessFactory;
+import org.geoserver.smartdataloader.metadata.jdbc.utils.JdbcUrlSplitter;
+import org.geotools.api.feature.type.Name;
+import org.geotools.data.postgis.PostgisNGDataStoreFactory;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.postgresql.jdbc.SslMode;
+
+/**
+ * Test class for testing the environment parametrization of the SmartDataLoader store. This test class set up a JDBC
+ * store with an env parametrized host, and use it from a SmartDataLoader store.
+ */
+public abstract class EnvParametrizationStoreTest extends AbstractJDBCSmartDataLoaderTestSupport {
+
+    private static final String SIMPLE_DATA_STORE_NAME = "simple-data-store";
+    private static final String SMART_DATA_LOADER_STORE_NAME = "smart-data-loader-store";
+    public static final String ALLOW_ENV_PARAMETRIZATION = "ALLOW_ENV_PARAMETRIZATION";
+    public static final String HOST_ENV_NAME = "TEST_DB_HOST";
+
+    private DataStoreInfo jdbcDataStore;
+
+    public EnvParametrizationStoreTest(JDBCFixtureHelper fixtureHelper) {
+        super(fixtureHelper);
+        NAMESPACE_PREFIX = "mt";
+        TARGET_NAMESPACE = "http://www.geo-solutions.it/smartappschema/1.0";
+    }
+
+    @BeforeClass
+    public static void testClassSetup() {
+        System.setProperty(ALLOW_ENV_PARAMETRIZATION, "true");
```

Comment:
I would also have a test where env parametrization is not enabled, to check there is no substitution